### PR TITLE
[8.0] Add hr timesheet hide to invoice

### PR DESCRIPTION
**Pull request by adrienpeiffer**
This module was written to add a security group to display invoicing rate field on timesheet line or not.
Therefore at the installation of this module, invoicing rate field is hidden for all users

### CODE REVIEW

File: hr_timesheet_invoice_hide_to_invoice/models/account_analytic_line.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with hr_timesheet_invoice_hide_to_invoice.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields
+
+GN = 'hr_timesheet_invoice_hide_to_invoice.group_invoice_rate_timesheet_line'
```
Comment (sbidoul) [inline hr_timesheet_invoice_hide_to_invoice/models/account_analytic_line.py]: I suppose the cryptic variable name is a side effect of pep8 complaints? :wink: 

Perhaps you can write ?

```
GROUP_ID = 'hr_timesheet_invoice_hide_to_invoice.
           'group_invoice_rate_timesheet_line')
```

**Comment by sbidoul** [comment]
This module covers a common requirement which is to prevent employees to decide by themselves if their timesheets are invoiceable or not.

:+1:

**Comment by lmignon** [comment]
I agree with @sbidoul for the variable name.
:+1:

**Comment by adrienpeiffer** [comment]
@sbidoul  It's done. Thanks! 
Indeed, it was because of a pep8 problem :smile:

**Comment by sbidoul** [comment]
This module is now working fine in production.

:+1: (functional test)

**Comment by sbidoul** [comment]
rebased

**Comment by sbidoul** [comment]
Is this one good for merging?

**Comment by adrienpeiffer** [comment]
yes :smile:

**Comment by ghost** [comment]
@sbidoul This module seams very specific to me. Is it for the needs of one of your clients or is it a behavior that is asked frequently to you ?

Also, you mask the field in the view, but the user still has access in read to the field.

**Comment by sbidoul** [comment]
@dufresnedavid this module is requested by all our clients of a certain size doing timesheets, where the employees enter timesheets but have no right to decide if it is invoiceable or not. And we use it at ACSONE too. So I guess it's a common requirement.

We tried to hide the field at the ORM level but this turned out to be impossible so we had to hide it in the view only.

File: hr_timesheet_invoice_hide_to_invoice/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#     hr_timesheet_invoice_hide_to_invoice is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with hr_timesheet_invoice_hide_to_invoice.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (ghost) [inline hr_timesheet_invoice_hide_to_invoice/__openerp__.py]: you can put the smaller version of the header here

File: hr_timesheet_invoice_hide_to_invoice_task/README.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+This module was written to add a security group to display invoicing rate field on task work or not.
+Therefore at the installation of this module, invoicing rate field is hidden for all users.1
+
+
+Installation
+============
+
+To install this module, you need to:
+
+ * Click on install button
```
Comment (ghost) [inline hr_timesheet_invoice_hide_to_invoice_task/README.rst]: is it really important to state that ?

**Comment by ghost** [comment]
@sbidoul ok, thanks, only 2 minor comments, otherwise LGTM.

**Comment by sbidoul** [comment]
@dufresnedavid thanks for the review. I changed the copyright header and clarified the module description in the README.

**Comment by ghost** [comment]
@sbidoul @adrienpeiffer  thanks for the contribution :+1:

**Pull request merged**